src/gui/shellwidget/test/CMakeLists.txt

```diff
@@ -6,6 +6,8 @@ endif ()
 
 function(add_xtest SOURCE_NAME)
 	add_executable(${SOURCE_NAME} ${SOURCE_NAME}.cpp)
+	file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR} TEST_SOURCE_DIR)
+	add_compile_definitions(${SOURCE_NAME} TEST_SOURCE_DIR="${TEST_SOURCE_DIR}")
```

Comment:
It seems `add_compile_definitions` does not work with older versions of cmake. This is why the tests are failing.

I think (untested) the alternative is to use `add_definitions(${SOURCE_NAME} -DTEST_SOURCE_DIR="${TEST_SOURCE_DIR}")`

Reply:
There's that, but also bench_widgetpaint segfaults on win64.

Reply:
That `add_definitions` line failed on linux. The following works fine, outside of the function:
`add_definitions(-DTEST_SOURCE_DIR="${TEST_SOURCE_DIR}")`
Since `add_definitions` only affects the directory and its subdirs I think that's fine.